core/src/main/java/io/grpc/HandlerRegistry.java

```diff
@@ -42,6 +44,16 @@
 public abstract class HandlerRegistry {
 
   /**
+   * Returns the {@link ServerServiceDefinition}s provided by the registry.
+   *
+   * @throws UnsupportedOperationException if not supported by implementation
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2222")
+  public List<ServerServiceDefinition> getServices() {
+    throw new UnsupportedOperationException("Not implemented");
```

Comment:
This exception is not being handled in `ServerImpl.getServices()`. Maybe it should just return `Collections.emptyList()`? May want to do the same with `Server`.